### ISSUE TITLE
Mizrahi scraper does not work in v0.9.0

### ISSUE THREAD

**Issue by philipgold** [eshaham/israeli-bank-scrapers#456]
hey @baruchiro . I see you're Mizrahi supporter.

I have a valid Mizrahi account. I'm trying to work with Mizrahi scraper. And it looks like broken.
I tested through [israeli-ynab-updater](https://github.com/eshaham/israeli-ynab-updater) and    mizrahi.test.js (israeli-bank-scrapers)

I'm using version - v0.9.0

I have the following errors

In israeli-ynab-updater :
```
scraping Mizrahi Bank
Mizrahi Bank: START_SCRAPING
Mizrahi Bank: INITIALIZING
Mizrahi Bank: LOGGING_IN
Mizrahi Bank: LOGIN_SUCCESS
Mizrahi Bank: TERMINATING
Mizrahi Bank: END_SCRAPING
success: false
error type: GENERIC
error: Timeout exceeded while waiting for event
```

In Mizrahi Tests resutls:
![image](https://content.screencast.com/users/philip.goldm/folders/Snagit/media/61a51000-44fb-4711-9838-828f7968b24a/2020-06-06_10-12-17.png
)

I will appreciate if you can help with that.

Thanks

**Comment by baruchiro** [comment]
This is about this line:
https://github.com/eshaham/israeli-bank-scrapers/blob/845f6c45682bf4afd256c918bca0bf28189d4a5c/src/scrapers/mizrahi.js#L114

The Osh page trigger an XHR request to get the transactions in JSON format, and we want this request.
But this request triggered immediately after the login, and this line waiting for a request that has already been made.

I don't remember if and how it worked before. Maybe this method can read requests a little back, or maybe at that time my Mizrahi homepage was another page than the Osh page, because of another issue.

I can't think how to solve this now, we should use some tricks in Puppeteer and I can't remember the API now.

For now, you can change your Mizrahi homepage.

**Comment by philipgold** [comment]
It looks like recently there were changed Home page with a new iframe Login page construction. Login process does not work anymore. 

![image](https://content.screencast.com/users/philip.goldm/folders/Snagit/media/07eedc18-72ab-4fb2-ac7b-951a5c164252/2020-06-08_08-50-48.png)

**Comment by baruchiro** [comment]
Right, we need to work on it.

In scrapers, I'm always trying to find the legacy websites, there are simpler and they are not changing. So I think we need to use this URL: https://www.mizrahi-tefahot.co.il/login/LoginMTO.aspx to login.

You're welcome to try to fix this.

**Comment by esakal** [comment]
@philipgold Hi, unfortunately I cannot assist as I don't have Mizrahi account. If you want to create a fix for that and need my assistance let me know.

**Comment by esakal** [comment]
@baruchiro  @philipgold any progress with this one?
Have a great weekend
Eran

**Comment by baruchiro** [comment]
Not from my side, sorry.

But @esakal , you can implement the login with the link I suggested (it also replaced, but still I think it's not going to change at the same rate as the home page), and send me the branch to run the tests.

I'm working on the [budget-tracking](https://github.com/brafdlog/budget-tracking), merging the [israeli-bank-scrapers-desktop](https://github.com/baruchiro/israeli-bank-scrapers-desktop), so I took a step back and got a lot of work.

**Comment by philipgold** [comment]
@esakal ,
I played with this. 

I stuck with submitting a button on the new Mizrahi form. [Code link](https://github.com/eshaham/israeli-bank-scrapers/blob/845f6c45682bf4afd256c918bca0bf28189d4a5c/src/scrapers/mizrahi.js#L106) 

I can't define how to call submit form button (`submitButtonSelector`) via element and selectors, due there is no id element. 

My code snippet:

```javascript
getLoginOptions(credentials) {
    return {
      loginUrl: `${LOGIN_URL}`,
      fields: createLoginFields(credentials),
      submitButtonSelector: 'btn.btn-primary.btn-block',
      postAction: async () => waitForNavigation(this.page, { waitUntil: 'networkidle0' }),
      possibleResults: getPossibleLoginResults(),
    };
```
HTML section in form:
```html
<div _ngcontent-cmw-c2="" class="form-group no-margin"><button _ngcontent-cmw-c2="" class="btn btn-primary btn-block"> כניסה </button></div>
```
It stuck in puppeteer's function `page.waitForSelector`. [Code link](http://github.com/eshaham/israeli-bank-scrapers/blob/845f6c45682bf4afd256c918bca0bf28189d4a5c/src/helpers/elements-interactions.js#L2-L2)


note:
* I found [a new login form,](https://www.mizrahi-tefahot.co.il/login/bv5x83ut/index.html#/auth-page-he) similar as old one. So no need to play with a new iframe Login page construction.

I can work on this. It will be great if I could consult with you about project details and some Node.js packages, such as `puppeteer`. I don't have deep Node.js knowledge. 

thanks 

cc: @baruchiro

**Comment by baruchiro** [comment]
Try this login page- https://sc.mizrahi-tefahot.co.il/SCServicesClientEZ/index.html#/login, I'm trying it now.
Seems that it sends the login credentials as simple JSON, but it has a bug because it not redirect you after the login.
You can navigate manually in the `postAction`, but I don't know if it is a good solution because, for example, I don't know it our `possibleResults` is supported in this login page.

But about the `submitButtonSelector`, you don't need to use only the `id` or the specific `class`es of the element. With *CSS Selectors* you can find by almost any structure. You can use the element name (`button`), you can use property and value (`button[type=submit]`).

About you link- https://www.mizrahi-tefahot.co.il/login/bv5x83ut/index.html#/auth-page-he, I'm opening this page in *Chrome* and open the *Developers Tools* (<kbd>F12</kbd>).
Then I <kbd>CTRL</kbd>+<kbd>F</kbd> in the *Elements* windows and I getting the Elements Search Box, you can search by CSS Selectors.
So let's start with the search `button`. You can see you have **two** buttons, looks the same. Why?
We looking at their fathers to see how to differentiate between them, because the CSS Selectors support a hierarchy path.
Try with `form` because we want a `button` in the `form`: `form button`. Still, we getting two buttons.
Now you can see these two buttons are in two different `div`s- one with `class="form-mobile"` and one with `class="form-desktop"`.
Try the selector `form .form-desktop button` and you get the expected button.

**Comment by baruchiro** [comment]
@philipgold I'm starting to work on it, any updates?

**Comment by esakal** [comment]
fixed in #500, currently available thru the Typescript branch (pending approval and merge in #404
